cardano-db-sync/src/Cardano/DbSync/Util/Whitelist.hs

```diff
@@ -0,0 +1,113 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Cardano.DbSync.Util.Whitelist where
+
+import Cardano.DbSync.Api.Types (InsertOptions (..), SyncEnv (..), SyncOptions (..))
+import Cardano.DbSync.Config.Types (MultiAssetConfig (..), PlutusConfig (..), ShelleyInsertConfig (..))
+import qualified Cardano.DbSync.Era.Shelley.Generic as Generic
+import Cardano.DbSync.Error (shortBsBase16Encode)
+import qualified Cardano.Ledger.Address as Ledger
+import qualified Cardano.Ledger.Credential as Ledger
+import Cardano.Ledger.Crypto (StandardCrypto)
+import Cardano.Ledger.Mary.Value (PolicyID (..))
+import Cardano.Prelude (NonEmpty)
+import Data.ByteString (ByteString)
+import Data.ByteString.Short (ShortByteString, toShort)
+import Data.Map (keys)
+
+-- check both whitelist but also checking plutus Maybes first
+plutusMultiAssetWhitelistCheck :: SyncEnv -> [Generic.TxOut] -> Bool
+plutusMultiAssetWhitelistCheck syncEnv txOuts =
+  isPlutusScriptHashesInWhitelist syncEnv txOuts || isMAPoliciesInWhitelist syncEnv txOuts
+
+isPlutusScriptHashesInWhitelist :: SyncEnv -> [Generic.TxOut] -> Bool
+isPlutusScriptHashesInWhitelist syncEnv txOuts = do
+  -- first check the config option
+  case ioPlutus iopts of
+    PlutusEnable -> True
+    PlutusDisable -> False
+    PlutusScripts plutusWhitelist -> plutuswhitelistCheck plutusWhitelist
+  where
+    iopts = soptInsertOptions $ envOptions syncEnv
+    plutuswhitelistCheck :: NonEmpty ShortByteString -> Bool
+    plutuswhitelistCheck whitelist =
+      any (\txOut -> isScriptHashWhitelisted whitelist txOut || isAddressWhitelisted whitelist txOut) txOuts
+    -- check if the script hash is in the whitelist
+    isScriptHashWhitelisted :: NonEmpty ShortByteString -> Generic.TxOut -> Bool
+    isScriptHashWhitelisted whitelist txOut =
+      maybe False ((`elem` whitelist) . toShort . Generic.txScriptHash) (Generic.txOutScript txOut)
+    -- check if the address is in the whitelist
+    isAddressWhitelisted :: NonEmpty ShortByteString -> Generic.TxOut -> Bool
+    isAddressWhitelisted whitelist txOut =
+      maybe False ((`elem` whitelist) . toShort) (Generic.maybePaymentCred $ Generic.txOutAddress txOut)
+
+isSimplePlutusScriptHashInWhitelist :: SyncEnv -> ByteString -> Bool
+isSimplePlutusScriptHashInWhitelist syncEnv scriptHash = do
+  case ioPlutus iopts of
+    PlutusEnable -> True
+    PlutusDisable -> False
+    PlutusScripts plutusWhitelist -> toShort scriptHash `elem` plutusWhitelist
```

Comment:
I think this should be just True. Whiletelisting is checked previoiusly in `isPlutusScriptHashesInWhitelist`. It also probably needs to be renamed to `isPlutusNotDisabled`.

Reply:
ok I'll check, but `isPlutusNotDisabled` is a double negative which in english makes it confusing

Reply:
@Cmdv thinking again, the current implementation may be correct. I thought the intented behaviout was different, but since the behaviour is open to interpretation, it's fine.

Reply:
So let's say we have a whitelist for a specific script A. Some tx contains script A in an output and script B in another output. The tx passes the validation check, but do we want to insert B? I guess both answers are correct as long as we are consinstent.